fix xatc eeprom settings

ellensp opened the pull request:
### Description

X_AXIS_TWIST_COMPENSATION incorrectly allocates space in SettingsDataStruct resulting in errors when writing data to eeprom 

### Requirements

X_AXIS_TWIST_COMPENSATION and EEPROM_SETTINGS

### Benefits

Simple things like M502 then M500 don't error 

### Related Issues
https://github.com/MarlinFirmware/Marlin/issues/23910
https://github.com/MarlinFirmware/Marlin/issues/23744#issuecomment-1069310196

ellensp committed:
fix xatc eeprom settings

Giuseppe499 commented:
Hi @ellensp,

I have some questions for you.

I don't know what this line does:
```
_FIELD_TEST(xatc);
```
If you know, could you explain it to me?

That line was added by @thinkyhead in PR #23745, now you edited it to be like this:
```diff
-_FIELD_TEST(xatc);
+_FIELD_TEST(xatc_spacing);
```
Why this edit was needed?

ellensp commented:
The test _FIELD_TEST references the variable in the structure SettingsDataStruct (structure that matches eeprom data byte for byte) 

I removed XATC xatc; as that is not correct and replaced it with the following that actually matches the code that reads ans writes to the eeprom latter on.

    float xatc_spacing;                                 // M423 X Z
    float xatc_start;
    xatc_array_t xatc_z_offset;

So  _FIELD_TEST cannot point to xatc as this no longer exists, it now is pointed it at the first variable of the new block.

thinkyhead commented:
Still weird that the XATC data is not read / written as expected. It shouldn't be related to byte alignment, as that will always be the same within the same build. Mysterious.
